docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -346,58 +387,72 @@ spec:
+
+##### RolloutType ScaleOut (create-swap-and-delete)
+
+Create-swap-and-delete rollout strategy is performed when `KubeadmControlPlane.Spec.RolloutStrategy` is not set or `KubeadmControlPlane.Spec.RolloutStrategy` is set to "ScaleOut".  
+
+- The rollout algorithm is the following:
+     - Find Machines that have an outdated spec
+     - If there is a machine requiring rollout
+        - Scale up control plane creating a machine with the new spec
+        - Scale down control plane by removing one of the machine that needs rollout (the oldest out-of date machine in the failure domain that has the most control-plane machines on it)
+
+##### RolloutType ScaleIn (using delete-and-create)
+
+Delete-and-create rollout strategy is performed when `KubeadmControlPlane.Spec.RolloutStrategy` is set to "ScaleIn". KCP rollout delete-and-create is recommended for resource constrained environment like bare-metal, OpenStack or vSphere resource pools, etc.
+
+- The algorithm is the following:
+  - Verify control plane replica count is >= 3
+    - If replica count is less than 3 inform the user that rollout can not be performed
+  - Scale down control plane by removing one of the machine that needs rollout
+  - Scale up control plane creating a machine with the new spec
+  - NOTE: control plane [Health checks](#Health checks) applies to scale up/scale down
 
 ###### Constraints and Assumptions
 
-* A stable endpoint (provided by DNS or IP) for the API server will be required in order to allow for machines to maintain a connection to control plane machines as they are swapped out during upgrades. This proposal is agnostic to how this is achieved, and is being tracked in https://github.com/kubernetes-sigs/cluster-api/issues/1687. The control plane controller will use the presence of the apiEndpoints status field of the cluster object to  determine whether or not to proceed. This behaviour is currently implicit in the implementations for cloud providers that provider a load balancer construct.
+  * A stable endpoint (provided by DNS or IP) for the API server will be required in order to allow for machines to maintain a connection to control plane machines as they are swapped out during upgrades. This proposal is agnostic to how this is achieved, and is being tracked in https://github.com/kubernetes-sigs/cluster-api/issues/1687. The control plane controller will use the presence of the apiEndpoints status field of the cluster object to  determine whether or not to proceed. This behaviour is currently implicit in the implementations for cloud providers that provider a load balancer construct.
+
+  * Infrastructure templates are expected to be immutable, so infrastructure template contents do not have to be hashed in order to detect
+    changes.
 
-* Infrastructure templates are expected to be immutable, so infrastructure template contents do not have to hashed in order to detect
-  changes.
-   
 ##### Remediation (using delete-and-recreate)
```

Comment:
What are the changes here? The diff isn't showing them properly, also could we split line 435 and limit the characters to ~80-100?

Reply:
I think this is caused by moving of the paragraph? We tried to arrange the proposal so that first we list the things that are common to both upgrade strategies and then in separate paragraphs individual features. So many of the paragraphs has been moved for one place to another.